src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -320,69 +326,251 @@ class DrawSketchHandlerLine: public DrawSketchHandlerLineBase
         return QString::fromLatin1("Sketcher_Pointer_Create_Line");
     }
 
+private:
+    ConstructionMethod constructionMethod;
 };
 
 // Function responsible for updating the DrawSketchHandler data members when widget parameters change
 template <> void DrawSketchHandlerLineBase::ToolWidgetManager::adaptDrawingToParameterChange(int parameterindex, double value) {

```

Comment:
This does not appear to be right.

The function needs to adapt the DSH to changes in the widget parameter. When "OnePointLengthAngle", it copies the length and angle into the points of the the edit curve (?). When it is the regular two point method and there is a change in the third and fourth parameters, nothing is updated (?)

Reply:
Ok. I have adapted this to the new architecture and is now in my branch.

Reply:
Not sure to get what you mean. Though if it is working as you implemented it I guess problem is solved.

Reply:
Eventually, you will need to take a look to the changes in architecture. That is also the reason why the commits cannot be blindly merged.

Reply:
Yes I'll definitely do it. Though now I'm on a tight spot and on top of that I just got covid so I'm very sleepy and can't do much.
I should have more time in 2 weeks or so.